pkg/security/podsecuritypolicy/provider.go

```diff
@@ -310,6 +316,10 @@ func (s *simpleProvider) ValidateContainerSecurityContext(pod *api.Pod, containe
 		}
 	}
 
+	if !s.psp.Spec.AllowPrivilegeEscalation && sc.AllowPrivilegeEscalation != nil && *sc.AllowPrivilegeEscalation {
```

Comment:
Let me see if I can describe the scenario better:

1. Admin sets `psp.AllowPrivilegeEscalation = false`, but leaves `psp.DefaultAllowPrivilegeEscalation = nil`.
2. User creates a pod with a non-root user, and leaves `pod.Spec.SecurityContext.AllowPrivilegeEscalation = nil`.
3. Validation check here lets it through, because it is nil.
4. Kubelet defaults it to allowed

As an admin, I would expect that the value I set in (1) would enforce that `AllowPrivilegeEscalation` is always false (ignoring the privileged pod case), and the result in (4) would surprise me.